### PR TITLE
feat(mito): make use of options in RegionCreate/OpenRequest

### PR DESCRIPTION
**Pull request by evenyag**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?
This PR stores `RegionOptions` from `RegionCreateRequest` and `RegionOpenRequest` in `Version`. Now we can get the options users specify while creating/opening the region.

Since `RegionCreateRequest` and `RegionOpenRequest` use `HashMap<String, String>` to transfer options, this PR serializes the map to a JSON string and deserializes it to the `RegionOptions` struct. Due to the limitation of [serde](https://github.com/serde-rs/serde/issues/1626), we still need to special handle nested structs.

This avoid writing lots of `get()` and `parse()` like this:
https://github.com/GreptimeTeam/greptimedb/blob/342cc0a4c4ba7dfa6fb8d202a142e9bf63ccd3a3/src/store-api/src/storage/engine.rs#L160-L194

### Refactor
- Renames `CompactionStrategy` to `CompactionOptions`
- Uses `Duration` to store compaction time window
- Moves `RegionOptions` to `options.rs`
- Removes `engine_options`

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
- #1869

### CODE REVIEW

**Review by fengjiachun — APPROVED** [review]
LGTM

File: src/mito2/src/compaction.rs [hunk 1]
```diff
@@ -176,8 +175,7 @@ impl CompactionScheduler {
     /// If the region has nothing to compact, it removes the region from the status map.
     fn schedule_compaction_request(&mut self, request: CompactionRequest) -> Result<()> {
         // TODO(hl): build picker according to region options.
```
Comment (killme2008) [inline src/mito2/src/compaction.rs:177]: Good job, why not fix `TODO` too?

File: src/mito2/src/worker/handle_open.rs [hunk 2]
```diff
@@ -62,6 +62,7 @@ impl<S: LogStore> RegionWorkerLoop<S> {
             self.scheduler.clone(),
         )
         .region_dir(&request.region_dir)
+        .options(request.options)
```
Comment (zhongzc) [inline src/mito2/src/worker/handle_open.rs:65]: Will there be a situation where the user repeatedly opens the region with different options?

Comment (evenyag) [reply src/mito2/src/worker/handle_open.rs:65]: Currently, reopening a region with different options is allowed.

File: src/mito2/src/compaction.rs (hunk 1 above)
Comment (evenyag) [reply src/mito2/src/compaction.rs:177]: Already fixed. I forgot to remove the todo.

Comment (evenyag) [inline src/mito2/src/compaction.rs:177]:
```suggestion
```

**Pull request merged**